app/helpers/taxonomy_helper.rb

```diff
@@ -135,4 +135,40 @@ def taxonomy_selects(f, selected_ids, taxonomy, label, options = {}, options_htm
     options[:label]    ||= _(label)
     multiple_selects f, label.downcase, taxonomy.authorized("assign_#{label.downcase}", taxonomy), selected_ids, options, options_html
   end
+
+  def all_checkbox(f, resource, label)
+    if User.current.admin? || User.current.filters.joins(:permissions).where({:'permissions.name' => "view_#{resource}",
+                                                       :search => nil,
+                                                       :taxonomy_search => nil}).present?
+      checkbox_f(f, :ignore_types,
+                   {:label => _("All") + label,
+                    :multiple => true,
+                    :onchange => 'ignore_checked(this)'},
+                    resource.humanize)
+    end
+  end
+
+  def show_resource_if_allowed(f, taxonomy, resource)
```

Comment:
I would do something like this:
``` Ruby
def show_resource_if_allowed(f, taxonomy, resource_options)
  if resource_options.is_a? Hash
    resource = resource_options[:resource]
    association = resource_options[:association]
  else
    resource = resource_options
    association = resource.classify.constantize
  end
  ...
```
It will remove the special cases from the function, and prepare the ground for more complex resource selection.